packages/app/src/systems/Settings/machines/setingsMachine.test.tsx

```diff
@@ -23,68 +22,43 @@ const WORDS = [
 ];
 
 describe('settingsMachine', () => {
-  let service: InterpreterFrom<typeof settingsMachine>;
-
   const redirectToWallet = jest.fn();
-
-  beforeAll(async () => {
-    jest
-      .spyOn(AccountService, 'exportVault')
-      .mockResolvedValue(WORDS.join(' '));
-  });
+  let service: InterpreterFrom<typeof settingsMachine>;
 
   beforeEach(async () => {
     service = interpret(
       settingsMachine.withConfig({
         actions: {
           goToWallet: redirectToWallet,
         },
+        services: {
+          exportVault: () => Promise.resolve(WORDS),
+          changePassword: () => Promise.resolve(),
+        },
       })
     ).start();
   });
 
   it('should get mnemonic phrase', async () => {
-    service.send('UNLOCK_WALLET', {
+    service.send('EXPORT_VAULT', {
       input: {
-        account: MOCK_ACCOUNTS[0],
         password: '123123',
       },
     });
+    await expectStateMatch(service, 'gettingMnemonic');
+    const state = await expectStateMatch(service, 'done');
 
-    await waitFor(service, (state) => state.matches('gettingMnemonic'));
-
-    const { matches } = await waitFor(service, (state) =>
-      state.matches('done')
-    );
-
-    const { event } = service.getSnapshot();
-
-    expect(event.data?.length).toBeGreaterThan(1);
-    expect(matches('done')).toBeTruthy();
+    expect(state.context.words.length).toBeGreaterThan(10);
   });
 
   it('should change the password of the user and we should be able to unlock it agan', async () => {
```

Comment:
```suggestion
  it('should change the password of the user', async () => {
```
The wallet should already be unlocked after we change the password so I think the test name can be changed.

Reply:
I add the tests to check if the password was updated thanks for the comment